packages/apiserver/src/resources/statuses/status.resolver.ts

```diff
@@ -7,24 +7,33 @@ import {
   Resolver,
 } from '@nestjs/graphql';
 import { Status } from './status.entity';
-import { BadRequestException, UseGuards } from '@nestjs/common';
+import {
+  BadRequestException,
+  ForbiddenException,
+  UseGuards,
+} from '@nestjs/common';
 import { CreateStatusInput } from '@mimir/global-types';
-import { AuthGuard } from '../../auth/auth.guard';
 import { Person } from '../persons/person.entity';
 import { StatusService } from './status.service';
 import { Material } from '../materials/material.entity';
+import { Role } from '../../auth/role.enum';
+import { CurrentUser } from '../../auth/current-user';
 
 @Resolver('Status')
 export class StatusResolver {
   constructor(private readonly statusService: StatusService) {}
   @Query(() => [Status])
-  @UseGuards(AuthGuard)
-  async getStatusesByPerson(@Args('person_id') id: string) {
+  async getStatusesByPerson(
+    @Args('person_id') id: string,
+    @CurrentUser() currentUser: Person
+  ) {
+    if (currentUser.type !== Role.Manager && +id !== +currentUser.id) {
+      return new ForbiddenException();
+    }
     return Status.find({ where: { person_id: id } });
   }
 
   @Query(() => [Status])
-  @UseGuards(AuthGuard)
```

Comment:
Why dou you remove all `AuthGuard` in this file? As I can see `Authguard` only uses here and nowhere else. From my perspective, `getStatusesByMaterial` as well as `createStatus` currently can be accessed via unauthorized person. All the application is connected to statuses, which means almost every request should have at least `id-token` . Did you test it via `graphql` playground if you are able to get access without any provided id?

Reply:
Auth guard registered globally via
`{ provide: APP_GUARD, useClass: AuthGuard },`
It will run for this query